x-pack/plugins/security_solution/server/lib/telemetry/sender.ts

```diff
@@ -281,6 +290,16 @@ export class TelemetryEventsSender {
 
     if (qlength >= this.maxQueueSize) {
       // we're full already
+      this.telemetryUsageCounter?.incrementCounter({
```

Comment:
This code seems to be a defence against a race condition. That is the queue wasn't drained and the poller has tried to put more docs in here. The code you want to gauge is a couple more lines down

```js
    if (events.length > this.maxQueueSize - qlength) {
      // <- in here
      this.queue.push(...this.processEvents(events.slice(0, this.maxQueueSize - qlength)));
    } else {
      this.queue.push(...this.processEvents(events));
    }
```

Reply:
Ah I see!